compiler/damlc/daml-stdlib-src/DA/Internal/Desugar.daml

```diff
@@ -106,10 +109,16 @@ Finally, the compiler has to unify the argument to `mkMethod` with the type
 -- Read: Interface `i` has a method named `m` with return type `r`
 class HasMethod i (m : Symbol) r | i m -> r
 
-newtype Method t i (m : Symbol) r = Method { unMethod : t -> r }
+-- | This is only a marker for the container, it doesn't contain the
+-- actual implementation of the method.
+newtype Method t i (m : Symbol) = Method ()
 
-mkMethod : (Implements t i, HasMethod i m r) => (t -> r) -> Method t i m r
-mkMethod = Method
+-- | This is used to check that the argument has the correct
+-- return type `r` for the given `t i m` type arguments.
+-- At runtime, the argument is ignored, but at compile time,
+-- it's extracted unmodified.
+mkMethod : (Implements t i, HasMethod i m r) => (t -> r) -> Method t i m
+mkMethod _ = Method ()
```

Comment:
Maybe `mkMethod` should actually be defined as "magic" like a lot of other functions that get erased (for example, GHC.Base's `getTag`). This would be another indicatio that we're erasing the call, and prevents us from accidentally not erasing `mkMethod` in the future.

```suggestion
mkMethod = magic @"mkMethod"
```

Reply:
that makes a lot of sense! I'll fix it here, I think it should be okay to leave the other `const (Method ())` definition in the GHC repo, right?

Reply:
yes, I think that works!